### PR TITLE
Forums gui

### PR DESCRIPTION
**Pull request by chelovechishko**

### CODE REVIEW

**Comment by chelovechishko** [comment]
Let's say it's no more [WIP]
Needs testing

**Comment by csoler** [comment]
Seems all good to me.

File: retroshare-gui/src/gui/gxsforums/GxsForumModel.cpp [hunk 1]
```diff
@@ -445,10 +445,13 @@ QVariant RsGxsForumModel::textColorRole(const ForumModelPostEntry& fmpe,int /*co
     if( (fmpe.mPostFlags & ForumModelPostEntry::FLAG_POST_IS_MISSING))
         return QVariant(mTextColorMissing);
 
-    if(IS_MSG_UNREAD(fmpe.mMsgStatus) || (fmpe.mPostFlags & ForumModelPostEntry::FLAG_POST_IS_PINNED))
+    if(IS_MSG_UNREAD(fmpe.mMsgStatus))
         return QVariant(mTextColorUnread);
     else
-        return QVariant(mTextColorRead);
+	    if(fmpe.mPostFlags & ForumModelPostEntry::FLAG_POST_IS_PINNED)
+		    return QVariant(mTextColorPinned);
```
Comment (csoler) [inline retroshare-gui/src/gui/gxsforums/GxsForumModel.cpp:452]: I dont know how that looks on your screen, but on mine, the pinned message becomes light orange over light pink background, so it's hardly readable. What was the expected effect?

File: retroshare-gui/src/gui/gxsforums/GxsForumThreadWidget.cpp [hunk 2]
```diff
@@ -1152,32 +1257,46 @@ void GxsForumThreadWidget::insertMessageData(const RsGxsForumMsg &msg)
...
+
+			ui->postText->setHtml(extraTxt) ;
+			return;
+		}
+		else {
+			RsIdentityDetails details;
+			rsIdentity->getIdDetails(msg.mMeta.mAuthorId, details);
+			QString name = GxsIdDetails::getName(details);
+
+			banned_text_info += "<p><font color=\"#e00000\"><b>" + tr( "The author of this message (with ID %1) is banned. And named by name ( %2 )").arg(QString::fromStdString(msg.mMeta.mAuthorId.toStdString()), name) + "</b>";
```
Comment (csoler) [inline retroshare-gui/src/gui/gxsforums/GxsForumThreadWidget.cpp:1277]: The sentence here is a bit weird. It should rather be "The author of this message (with ID %1 and name "%2") is banned."

File: retroshare-gui/src/gui/qss/stylesheet/qss.default [hunk 3]
```diff
@@ -138,6 +138,7 @@ ForumsDialog, GxsForumThreadWidget
 	qproperty-textColorUnreadChildren: darkgray;
 	qproperty-textColorNotSubscribed: black;
 	qproperty-textColorMissing: darkRed;
+	qproperty-textColorPinned: darkOrange;
```
Comment (chelovechishko) [inline retroshare-gui/src/gui/qss/stylesheet/qss.default:141]: Of course somebody has to change it to what ordinary instance could use as usable. I thought background is plain white and darkorange was not an unwise choice.
my_colors
default_qss_colors

File: retroshare-gui/src/gui/gxsforums/GxsForumThreadWidget.cpp (hunk 2 above)
Comment (chelovechishko) [reply retroshare-gui/src/gui/gxsforums/GxsForumThreadWidget.cpp:1277]: English is not my native language.

File: retroshare-gui/src/gui/gxsforums/GxsForumModel.cpp (hunk 1 above)
Comment (chelovechishko) [reply retroshare-gui/src/gui/gxsforums/GxsForumModel.cpp:452]: Paint it black?

Comment (NyvenZA) [reply retroshare-gui/src/gui/gxsforums/GxsForumModel.cpp:452]: Have the same problem:
image

Comment (chelovechishko) [reply retroshare-gui/src/gui/gxsforums/GxsForumModel.cpp:452]: 'cause color is [still] darkorange i would be dazed and highly discouraged if you would saw something different.

**Comment by chelovechishko** [comment]
Also to paint it black :
[0001-qss-gxsforums-change-text-color-of-pinned-posts-to-b.patch.txt](https://github.com/RetroShare/RetroShare/files/5262136/0001-qss-gxsforums-change-text-color-of-pinned-posts-to-b.patch.txt)

**Pull request merged**